util/clusterauth/clusterauth.go

```diff
@@ -258,24 +258,11 @@ func GetServiceAccountBearerToken(clientset kubernetes.Interface, ns string, sa
 func getOrCreateServiceAccountTokenSecret(clientset kubernetes.Interface, sa, ns string) (string, error) {
 	// Wait for sa to have secret, but don't wait too
 	// long for 1.24+ clusters
-	var serviceAccount *corev1.ServiceAccount
-	err := wait.PollImmediate(500*time.Millisecond, 5*time.Second, func() (bool, error) {
-		ctx, cancel := context.WithTimeout(context.Background(), common.ClusterAuthRequestTimeout)
-		defer cancel()
-		var getErr error
-		serviceAccount, getErr = clientset.CoreV1().ServiceAccounts(ns).Get(ctx, sa, metav1.GetOptions{})
-		if getErr != nil {
-			return false, fmt.Errorf("failed to get serviceaccount %q: %w", sa, getErr)
-		}
-		if len(serviceAccount.Secrets) == 0 {
-			return false, nil
-		}
-		return true, nil
-	})
-	if err != nil && err != wait.ErrWaitTimeout {
-		return "", fmt.Errorf("failed to get serviceaccount token secret: %w", err)
+	serviceAccount, err := clientset.CoreV1().ServiceAccounts(ns).Get(context.Background(), sa, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to get serviceaccount %q: %w", sa, err)
```

Comment:
I'm surprised to see the removal of the poll mechanism here. What's the intent?

Reply:
Ah interesting. Okay, wanna go ahead and revert it? I think we're past the codegen issues.

Reply:
Okay, I have a bit question. Is the problem related with codegen issue?
If so, I agree to revert it.
If you feel free to revert it, could you do it ?

Reply:
I reverted it.

Reply:
Nah, should be unrelated. Thanks!